src/nvim/os/input.c

```diff
@@ -188,8 +188,13 @@ size_t input_enqueue(String keys)
   char *ptr = keys.data;
   char *end = ptr + keys.size;
 
-  while (rbuffer_space(input_buffer) >= 6 && ptr < end) {
-    uint8_t buf[6] = { 0 };
+  while (rbuffer_space(input_buffer) >= 19 && ptr < end) {
+    // A "<x>" form occupies at least 1 characters, and produces up
+    // to 19 characters (1 + 5 * 3 for the char and 3 for a modifier).
+    // In the case of K_SPECIAL(0x80) or CSI(0x9B), 3 bytes are escaped and
+    // needed, but since the keys are UTF-8, so the first byte cannot be
+    // K_SPECIAL(0x80) or CSI(0x9B).
+    uint8_t buf[19] = { 0 };
```

Comment:
I believe we should use MB_MAXBYTES + 1 as that's mandated by `add_char2buf`. You seem to know a lot about the encoding stuff and I fail at understanding in what situation we can have ` one 16-bit character of up to three bytes plus six following composing characters of three bytes each.` at https://github.com/neovim/neovim/blob/fc98f2d5815d8f7e671766db255edbb7365960b1/src/nvim/vim.h#L287 do you have any idea ?

Reply:
> You seem to know a lot about the encoding stuff and I fail at understanding in what situation we can have one 16-bit character of up to three bytes plus six following composing characters of three bytes each. at ... do you have any idea ?

I'm not particularly familiar with encoding. I looked at the other parts of using MB_MAXBYTES, but I couldn't see when the situation would be similar to the one in the comment.

Reply:
> I believe we should use `MB_MAXBYTES + 1` as that's mandated by `add_char2buf`.

I think the comment on `add_char2buf` is wrong. `add_char2buf` allocates a buffer of `MB_MAXBYTES+1`, but `utf_char2bytes` only uses a maximum of 6 bytes. The first byte is never `K_SPECIAL(0x80)`, so the maximum buffer needed by `add_char2buf` is `3 * 5 + 1 = 16` bytes.

Reply:
Agreed.